File: app/src/main/java/com/lepitar/corvid19/FinsihDialog.java
package com.lepitar.corvid19;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.initialization.InitializationStatus;
import com.google.android.gms.ads.initialization.OnInitializationCompleteListener;

import static android.content.Context.MODE_PRIVATE;

public class FinsihDialog extends Dialog implements View.OnClickListener {
    private TextView btn_cancel;
    private TextView btn_ok;
    private TextView text_alarm;
    private AdView adView;

    Context context;

    public FinsihDialog(@NonNull Context context) {
        super(context);
        this.context = context;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.finish_dialog);

        btn_cancel = (TextView) findViewById(R.id.btn_cancel);
        btn_ok = (TextView) findViewById(R.id.btn_ok);
        text_alarm = findViewById(R.id.text_dialog);

        MobileAds.initialize(context, new OnInitializationCompleteListener() {
            @Override
            public void onInitializationComplete(InitializationStatus initializationStatus) {
            }
        });
        adView = findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        adView.loadAd(adRequest);

        btn_cancel.setOnClickListener(this);
        btn_ok.setOnClickListener(this);
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_cancel:
                dismiss();
                break;
            case R.id.btn_ok:
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    ((Activity) context).finishAndRemoveTask();
                }
                SharedPreferences.Editor addeditor = context.getSharedPreferences("add", MODE_PRIVATE).edit();
                addeditor.putBoolean("add", false);
                addeditor.apply();
                ((Activity) context).finishAffinity();
                System.exit(0);
                break;
        }
    }
}

File: app/src/main/java/com/lepitar/corvid19/FinsihDialog.kt
package com.lepitar.corvid19

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdView
import com.google.android.gms.ads.MobileAds
import kotlinx.android.synthetic.main.finish_dialog.*
import kotlin.system.exitProcess

class FinsihDialog(context: Context) : Dialog(context), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.finish_dialog)
        MobileAds.initialize(context) { }
        val adRequest = AdRequest.Builder().build()
        adView.loadAd(adRequest)
        btn_cancel!!.setOnClickListener(this)
        btn_ok!!.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_cancel -> dismiss()
            R.id.btn_ok -> {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    ownerActivity?.finishAndRemoveTask()
                }
                ownerActivity?.finishAffinity()
                ownerActivity?.moveTaskToBack(true);
                exitProcess(1)
            }
        }
    }

}